helpers.go

```diff
@@ -267,29 +268,111 @@ func getOffer(header string, isAccepted func(spec, offer string) bool, offers ..
 		return offers[0]
 	}
 
-	for _, offer := range offers {
-		if len(offer) == 0 {
-			continue
+	type acceptedType struct {
+		spec        string
+		quality     float64
+		specificity int
+		order       int
+	}
+
+	// Parse header and get accepted types with their quality and specificity
+	// See: https://www.rfc-editor.org/rfc/rfc9110#name-content-negotiation-fields
+	spec, commaPos, order := "", 0, 0
+	acceptedTypes := make([]acceptedType, 0, 20)
```

Comment:
can we use syncPool to optimize the initialization?

Reply:
It's slower and adds 2 allocs:

syncPool:

```console
~/Documents/GitHub/sixcolors/fiber (2387-ctx-accepts-get-offer-q-and-specificity*) » go test -v -run=^$ -bench=Benchmark_Ctx_Accepts -benchmem -count=4
goos: darwin
goarch: arm64
pkg: github.com/gofiber/fiber/v2
Benchmark_Ctx_Accepts
Benchmark_Ctx_Accepts/run-[]string{".xml"}
Benchmark_Ctx_Accepts/run-[]string{".xml"}-12            3662680               308.6 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{".xml"}-12            3888830               307.4 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{".xml"}-12            3836806               306.3 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{".xml"}-12            3864616               308.3 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}-12     3338768               350.2 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}-12     3397196               351.2 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}-12     3334743               353.5 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}-12     3366334               352.8 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}-12             3876166               310.0 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}-12             3752412               307.9 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}-12             3703234               309.1 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}-12             3796650               311.3 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsCharsets
Benchmark_Ctx_AcceptsCharsets-12                                                         4684959               251.0 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsCharsets-12                                                         4563364               254.2 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsCharsets-12                                                         4649942               251.3 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsCharsets-12                                                         4701276               252.6 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsEncodings
Benchmark_Ctx_AcceptsEncodings-12                                                        4154277               282.1 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsEncodings-12                                                        4155589               283.1 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsEncodings-12                                                        4226884               285.7 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsEncodings-12                                                        4220952               284.2 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsLanguages
Benchmark_Ctx_AcceptsLanguages-12                                                        3340310               350.9 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsLanguages-12                                                        3328549               350.6 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsLanguages-12                                                        3369895               353.3 ns/op           920 B/op          2 allocs/op
Benchmark_Ctx_AcceptsLanguages-12                                                        3324868               352.1 ns/op           920 B/op          2 allocs/op
PASS
ok      github.com/gofiber/fiber/v2     36.191s
```

vs `acceptedTypes := make([]acceptedType, 0, 20)`:

```console
goos: darwin
goarch: arm64
pkg: github.com/gofiber/fiber/v2
Benchmark_Ctx_Accepts
Benchmark_Ctx_Accepts/run-[]string{".xml"}
Benchmark_Ctx_Accepts/run-[]string{".xml"}-12            7756831               141.6 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{".xml"}-12            8427858               140.9 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{".xml"}-12            8356838               156.3 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{".xml"}-12            8602822               139.1 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}-12     6509782               183.1 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}-12     6445286               184.8 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}-12     6444319               184.9 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"json",_"xml"}-12     6491694               185.3 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}-12             8612815               139.2 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}-12             8531341               139.0 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}-12             8657082               141.0 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_Accepts/run-[]string{"application/json",_"application/xml"}-12             8553763               139.5 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsCharsets
Benchmark_Ctx_AcceptsCharsets-12                                                        14816422                74.18 ns/op            0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsCharsets-12                                                        16124913                81.23 ns/op            0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsCharsets-12                                                        15506538                80.20 ns/op            0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsCharsets-12                                                        15934032                75.11 ns/op            0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsEncodings
Benchmark_Ctx_AcceptsEncodings-12                                                       10669336               108.5 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsEncodings-12                                                       10658364               117.1 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsEncodings-12                                                       10817725               108.8 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsEncodings-12                                                       10624474               108.7 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsLanguages
Benchmark_Ctx_AcceptsLanguages-12                                                        6412471               181.4 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsLanguages-12                                                        6546268               180.5 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsLanguages-12                                                        6572288               180.2 ns/op             0 B/op          0 allocs/op
Benchmark_Ctx_AcceptsLanguages-12                                                        6550489               183.7 ns/op             0 B/op          0 allocs/op
PASS
ok      github.com/gofiber/fiber/v2     32.400s
```